pkg/hostagent/setup.go

```diff
@@ -284,6 +286,16 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 			}
 		}
 
+		for _, ip := range iface.IPs {
+			if ip.PodIp != nil {
```

Comment:
Do we really need a separate PodIp field? My reading of the code is that the existing Address.IP field will get set with the allocated address already. Since you do the IP allocate first now, it will only get this far if an IP has been allocated

Reply:
Yes, you are right.  we don't need a separate field. I was under as assumption that the existing Address field is of type net.IPNet which is a combination of address and mask. whereas ip.SetHWAddrByIP takes the IP parameter which is of type "net.IP". Since we can directly access Address.IP , we wouldn't need this extra field.